modules/utxo-lib/src/bitgo/UtxoPsbt.ts

```diff
@@ -217,6 +217,29 @@ export class UtxoPsbt<Tx extends UtxoTransaction<bigint> = UtxoTransaction<bigin
     return this;
   }
 
+  finalizeTapInputWithSingleLeafScriptAndSignature(inputIndex: number): this {
+    const input = checkForInput(this.data.inputs, inputIndex);
+    if (input.tapLeafScript?.length !== 1) {
+      throw new Error('Only one leaf script supported for finalizing');
+    }
+    if (!input.tapScriptSig) {
```

Comment:
`input.tapScriptSig?.length !== 1`

Reply:
technically we "should" make sure that the script sig is for the same script as we found above, but DGAF for this since we know how it's gonna be used.